src/main/java/wooteco/subway/line/LineH2Dao.java

```diff
@@ -0,0 +1,86 @@
+package wooteco.subway.line;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Repository
+public class LineH2Dao implements LineRepository {
+    private final JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public LineH2Dao(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private final RowMapper<Line> lineRowMapper = (rs, rn) -> {
+        long id = rs.getLong("id");
+        String name = rs.getString("name");
+        String color = rs.getString("color");
+        return new Line(id, name, color);
+    };
+
+    @Override
+    public Line save(Line line) {
+        String query = "INSERT INTO LINE (name, color) VALUES (?, ?)";
+
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        this.jdbcTemplate.update(connection -> {
+            PreparedStatement ps = connection
+                    .prepareStatement(query, new String[]{"id"});
+            ps.setString(1, line.getName());
+            ps.setString(2, line.getColor());
+            return ps;
+        }, keyHolder);
+
+        return new Line(Objects.requireNonNull(keyHolder.getKey()).longValue(), line.getName(), line.getColor());
```

Comment:
```suggestion
        return findById(keyHolder.getKey());
```
와 같이 표현하는것은 어떻게 생각하시나요?

Reply:
[좋은 제안](https://github.com/woowacourse/atdd-subway-map/pull/95/commits/0607be447f254dce9874c5082d4aec852ae1c09b)이십니다!! 😀🙌